### PR TITLE
Order dependencies alphabetically in multiple formulae as per `brew audit`

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] ~Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.~

-----

Some failures remain, but here's the fixes for to `brew audit --only-cops=DependencyOrder` (and a small unrelated one in the final commit as it was too easy to not fix when I ran the required tests on all formulae). These didn't appear when I ran the `brew audit` command in Homebrew, hence submitting them here.

### CODE REVIEW

**Comment by iMichka** [comment]
Thanks for working on this. This is indeed a much needed cleanup.

I am not sure all of these are linux-only changes. Some may need to be fixed upstream (in the core mac repo first).

Could you split this a little bit to ease the review? Maybe just the `depends_on CIRequirement` and 
`depends_on :macos` first. I am sorry that you will have to redo the changes again, but it will help me reviewing this.

**Comment by issyl0** [comment]
Thanks @iMichka for the review. I thought that some of them belonged in Homebrew too, but running `brew audit --only-cops=DepedencyOrder` on my Mac on an updated master didn't show anything. 😕

Nevertheless, I'll check again later today after work and back some of the changes out so it's only `depends_on CIRequirement` and `depends_on :macos`. Re-doing it isn't a problem, this PR needs a rebase anyway!

**Comment by issyl0** [comment]
33 commits are now 14, and they're only changes for `CIRequirement` and `macos`. Hopefully this is more manageable!

File: Formula/duck.rb [hunk 1]
```diff
@@ -13,11 +13,11 @@ class Duck < Formula
     sha256 "907976c6f876a6e811ef61657af793578caaf51fdc400a38a3569bff6c75d370" => :sierra
   end
 
-  depends_on :macos
   depends_on "ant" => :build
   depends_on :java => ["1.8", :build]
   depends_on "maven" => :build
   depends_on :xcode => :build
+  depends_on :macos
```
Comment (issyl0) [inline Formula/duck.rb]: Oops, this is a duplicate (and also in the wrong place). Fixed.

**Comment by iMichka** [comment]
Thanks @issyl0 !

**Pull request closed without merging**